### ISSUE TITLE
Roslyn crashes on VS2015 Update3 RC

### ISSUE THREAD

**Issue by bkoelman** [dotnet/roslyn#12066]
**Version Used**: 
VS2015 Update3 RC, master branch at https://github.com/dotnet/roslyn/commit/5416d9c8111a39e3f7a84f52cfb4d9123e1328e5.

**Steps to Reproduce**:

1. Follow steps at [Building, Debugging, and Testing on Windows](https://github.com/dotnet/roslyn/blob/master/docs/contributing/Building,%20Debugging,%20and%20Testing%20on%20Windows.md), set VisualStudioSetup as startup project, F5
2. In second VS instance, create new C# Console Application
3. First instance with debugger attached breaks with exception in [IncrementalAnalyzerProcessor](https://github.com/dotnet/roslyn/blob/master/src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.IncrementalAnalyzerProcessor.cs#L100):

```
An exception of type 'System.TypeLoadException' occurred in Microsoft.CodeAnalysis.Features.dll but was not handled in user code

Additional information: Method 'AnalyzeSyntaxAsync' in type 'Analyzer' from assembly 'Microsoft.VisualStudio.LanguageServices.Telemetry, Version=1.3.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' does not have an implementation.
```
4. Detach debugger; second VS instance crashes.

**Expected Behavior**:
Project gets created and loaded successfully.

**Actual Behavior**:
VS crashes.

**Comment by bkoelman** [comment]
During build, VS shows the next warnings in Error list that might be related:

```
Severity	Code	Description	Project	File	Line	Suppression State
Warning		Unable to add a reference to project 'BasicScripting'. The targets of Portable Library project 'BasicScripting' are not the same or compatible with the targets of the current Portable Library project.

A Portable Library project's targets can be changed via the Library tab in the project's properties.	ScriptingTest			
Warning		Unable to add a reference to project 'BasicScripting'. The targets of Portable Library project 'BasicScripting' are not the same or compatible with the targets of the current Portable Library project.

A Portable Library project's targets can be changed via the Library tab in the project's properties.	BasicScriptingTest			
Warning		Unable to add a reference to project 'BasicScripting'. The targets of Portable Library project 'BasicScripting' are not the same or compatible with the targets of the current Portable Library project.

A Portable Library project's targets can be changed via the Library tab in the project's properties.	VbiCore			
Warning		Unable to add a reference to project 'CommonCoreClrRuntime'. The targets of Portable Library project 'CommonCoreClrRuntime' are not the same or compatible with the targets of the current Portable Library project.

A Portable Library project's targets can be changed via the Library tab in the project's properties.	DeployCompilerGeneratorToolsRuntime			
Warning		Unable to add a reference to project 'CommonNetCoreReferences'. The targets of Portable Library project 'CommonNetCoreReferences' are not the same or compatible with the targets of the current Portable Library project.

A Portable Library project's targets can be changed via the Library tab in the project's properties.	CscCore			
Warning		Unable to add a reference to project 'CommonNetCoreReferences'. The targets of Portable Library project 'CommonNetCoreReferences' are not the same or compatible with the targets of the current Portable Library project.

A Portable Library project's targets can be changed via the Library tab in the project's properties.	CsiCore			
Warning		Unable to add a reference to project 'CommonNetCoreReferences'. The targets of Portable Library project 'CommonNetCoreReferences' are not the same or compatible with the targets of the current Portable Library project.

A Portable Library project's targets can be changed via the Library tab in the project's properties.	DeployCoreClrTestRuntime			
Warning		Unable to add a reference to project 'CommonNetCoreReferences'. The targets of Portable Library project 'CommonNetCoreReferences' are not the same or compatible with the targets of the current Portable Library project.

A Portable Library project's targets can be changed via the Library tab in the project's properties.	PortableServer			
Warning		Unable to add a reference to project 'CommonNetCoreReferences'. The targets of Portable Library project 'CommonNetCoreReferences' are not the same or compatible with the targets of the current Portable Library project.

A Portable Library project's targets can be changed via the Library tab in the project's properties.	VbcCore			
Warning		Unable to add a reference to project 'CommonNetCoreReferences'. The targets of Portable Library project 'CommonNetCoreReferences' are not the same or compatible with the targets of the current Portable Library project.

A Portable Library project's targets can be changed via the Library tab in the project's properties.	VbiCore			
Warning		Unable to add a reference to project 'CompilersBoundTreeGenerator'. The targets of Portable Library project 'CompilersBoundTreeGenerator' are not the same or compatible with the targets of the current Portable Library project.

A Portable Library project's targets can be changed via the Library tab in the project's properties.	DeployCompilerGeneratorToolsRuntime			
Warning		Unable to add a reference to project 'CSharpCompilerSyntaxTest'. The targets of Portable Library project 'CSharpCompilerSyntaxTest' are not the same or compatible with the targets of the current Portable Library project.

A Portable Library project's targets can be changed via the Library tab in the project's properties.	DeployCoreClrTestRuntime			
Warning		Unable to add a reference to project 'CSharpCompilerTestUtilities'. The targets of Portable Library project 'CSharpCompilerTestUtilities' are not the same or compatible with the targets of the current Portable Library project.

A Portable Library project's targets can be changed via the Library tab in the project's properties.	CSharpCompilerSyntaxTest			
Warning		Unable to add a reference to project 'CSharpErrorFactsGenerator'. The targets of Portable Library project 'CSharpErrorFactsGenerator' are not the same or compatible with the targets of the current Portable Library project.

A Portable Library project's targets can be changed via the Library tab in the project's properties.	DeployCompilerGeneratorToolsRuntime			
Warning		Unable to add a reference to project 'CSharpScripting'. The targets of Portable Library project 'CSharpScripting' are not the same or compatible with the targets of the current Portable Library project.

A Portable Library project's targets can be changed via the Library tab in the project's properties.	ScriptingTest			
Warning		Unable to add a reference to project 'CSharpScripting'. The targets of Portable Library project 'CSharpScripting' are not the same or compatible with the targets of the current Portable Library project.

A Portable Library project's targets can be changed via the Library tab in the project's properties.	CSharpScriptingTest			
Warning		Unable to add a reference to project 'CSharpScripting'. The targets of Portable Library project 'CSharpScripting' are not the same or compatible with the targets of the current Portable Library project.

A Portable Library project's targets can be changed via the Library tab in the project's properties.	CsiCore			
Warning		Unable to add a reference to project 'CSharpSyntaxGenerator'. The targets of Portable Library project 'CSharpSyntaxGenerator' are not the same or compatible with the targets of the current Portable Library project.

A Portable Library project's targets can be changed via the Library tab in the project's properties.	DeployCompilerGeneratorToolsRuntime			
Warning		Unable to add a reference to project 'Scripting'. The targets of Portable Library project 'Scripting' are not the same or compatible with the targets of the current Portable Library project.

A Portable Library project's targets can be changed via the Library tab in the project's properties.	BasicScripting			
Warning		Unable to add a reference to project 'Scripting'. The targets of Portable Library project 'Scripting' are not the same or compatible with the targets of the current Portable Library project.

A Portable Library project's targets can be changed via the Library tab in the project's properties.	CSharpScripting			
Warning		Unable to add a reference to project 'Scripting'. The targets of Portable Library project 'Scripting' are not the same or compatible with the targets of the current Portable Library project.

A Portable Library project's targets can be changed via the Library tab in the project's properties.	ScriptingTest			
Warning		Unable to add a reference to project 'Scripting'. The targets of Portable Library project 'Scripting' are not the same or compatible with the targets of the current Portable Library project.

A Portable Library project's targets can be changed via the Library tab in the project's properties.	BasicScriptingTest			
Warning		Unable to add a reference to project 'Scripting'. The targets of Portable Library project 'Scripting' are not the same or compatible with the targets of the current Portable Library project.

A Portable Library project's targets can be changed via the Library tab in the project's properties.	CSharpScriptingTest			
Warning		Unable to add a reference to project 'Scripting'. The targets of Portable Library project 'Scripting' are not the same or compatible with the targets of the current Portable Library project.

A Portable Library project's targets can be changed via the Library tab in the project's properties.	CsiCore			
Warning		Unable to add a reference to project 'Scripting'. The targets of Portable Library project 'Scripting' are not the same or compatible with the targets of the current Portable Library project.

A Portable Library project's targets can be changed via the Library tab in the project's properties.	VbiCore			
Warning		Unable to add a reference to project 'ScriptingTest'. The targets of Portable Library project 'ScriptingTest' are not the same or compatible with the targets of the current Portable Library project.

A Portable Library project's targets can be changed via the Library tab in the project's properties.	BasicScriptingTest			
Warning		Unable to add a reference to project 'ScriptingTest'. The targets of Portable Library project 'ScriptingTest' are not the same or compatible with the targets of the current Portable Library project.

A Portable Library project's targets can be changed via the Library tab in the project's properties.	CSharpScriptingTest			
Warning		Unable to add a reference to project 'ScriptingTest'. The targets of Portable Library project 'ScriptingTest' are not the same or compatible with the targets of the current Portable Library project.

A Portable Library project's targets can be changed via the Library tab in the project's properties.	DeployCoreClrTestRuntime			
Warning		Unable to add a reference to project 'TestUtilities'. The targets of Portable Library project 'TestUtilities' are not the same or compatible with the targets of the current Portable Library project.

A Portable Library project's targets can be changed via the Library tab in the project's properties.	CSharpCompilerTestUtilities			
Warning		Unable to add a reference to project 'TestUtilities'. The targets of Portable Library project 'TestUtilities' are not the same or compatible with the targets of the current Portable Library project.

A Portable Library project's targets can be changed via the Library tab in the project's properties.	ScriptingTest			
Warning		Unable to add a reference to project 'TestUtilities'. The targets of Portable Library project 'TestUtilities' are not the same or compatible with the targets of the current Portable Library project.

A Portable Library project's targets can be changed via the Library tab in the project's properties.	BasicScriptingTest			
Warning		Unable to add a reference to project 'TestUtilities'. The targets of Portable Library project 'TestUtilities' are not the same or compatible with the targets of the current Portable Library project.

A Portable Library project's targets can be changed via the Library tab in the project's properties.	CSharpCompilerSyntaxTest			
Warning		Unable to add a reference to project 'TestUtilities'. The targets of Portable Library project 'TestUtilities' are not the same or compatible with the targets of the current Portable Library project.

A Portable Library project's targets can be changed via the Library tab in the project's properties.	CSharpScriptingTest			
Warning		Unable to add a reference to project 'VisualBasicErrorFactsGenerator'. The targets of Portable Library project 'VisualBasicErrorFactsGenerator' are not the same or compatible with the targets of the current Portable Library project.

A Portable Library project's targets can be changed via the Library tab in the project's properties.	DeployCompilerGeneratorToolsRuntime			
Warning		Unable to add a reference to project 'VisualBasicSyntaxGenerator'. The targets of Portable Library project 'VisualBasicSyntaxGenerator' are not the same or compatible with the targets of the current Portable Library project.

A Portable Library project's targets can be changed via the Library tab in the project's properties.	DeployCompilerGeneratorToolsRuntime			
```

**Comment by Pilchie** [comment]
Tag @heejaechang

**Comment by heejaechang** [comment]
tag @KevinH-MS

**Comment by jmarolf** [comment]
We consider them "public" if they break the public development experience, yes.

**Comment by heejaechang** [comment]
Unless we provide them a way to patch those side as well…


-          heejae

**Comment by jmarolf** [comment]
@heejaechang I don't particularly care about *how* we go about it, as long as when someone presses F5 everything works.

**Comment by heejaechang** [comment]
Yay, I agree we should make people be able to run vs using their own bit, but I just hoped that doesn’t involve making cross cutting changes harder than now.


-          heejae

**Comment by KevinH-MS** [comment]
Fixed with #12074.  Thanks for reporting and sorry about that!

**Comment by bkoelman** [comment]
This was only the first in a series of errors. The next one is at https://github.com/dotnet/roslyn/blob/master/src/VisualStudio/CSharp/Impl/Options/AutomationObject.cs#L660:

```
An exception of type 'System.TypeLoadException' occurred in mscorlib.dll but was not handled in user code

Additional information: Method 'AnalyzeSyntaxAsync' in type 'Analyzer' from assembly 'Microsoft.VisualStudio.LanguageServices.Telemetry, Version=1.3.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' does not have an implementation.
```

And then later followed by:
![image](https://cloud.githubusercontent.com/assets/10324372/16178568/88f903ec-364c-11e6-8a6b-39a78d11c6eb.png)
